instrumentedProfilingLibrary/Time Profiler/ProfileDataManager.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ProfileDataManager.h"
+#include <iostream>
+
+using namespace std;
+
+ProfileDataManager* ProfileDataManager::profiledatamanager = nullptr;
+
+ProfileDataManager::ProfileDataManager(const string name){

```

Comment:
Some problems here:
* passing strings by value is inefficient as it copies memory (usually!) - if you just need to _read_ the string data, prefer passing by reference. Copying can be avoided by using move semantics, but that invalidates the string of the caller scope. For simplicity, if you can read it without any side-effects, pass by reference (const reference to ensure you don't modify it inside the function)

* the `name` parameter is not actually used

* the file is directly open in the constructor. This has several side-effects: you can end up throwing an exception (cannot open/write to disk etc.), then you keep the file open for the duration of the process. Do you need o keep it alive?

Reply:
I updated `ProfileDataManager::ProfileDataManager(const string name)`  to  `ProfileDataManager::ProfileDataManager()` since the the string value `name`  was not necessary/not used. Concerning the passing by value/by reference and the memory efficiency, I will keep that in mind.
Concerning the file opening in the constructor, I had initially thought it would be a good idea to keep it open until the end of the program to store all the code blocks time profiles, I have re-written it with the file opening and closing in my main cpp file and making the file accessible to the` ProfileDataManager`.